mod - updating required mobile version.

mebassett opened the pull request:
this will force yoroi into maint mode for all users until we are
happy with a shelley-enabled version.

Commit:
mod - updating required mobile version.

this will force yoroi into maint mode for all users until we are
happy with a shelley-enabled version.

nicarq commented on src/index.ts:
```diff
@@ -241,7 +241,7 @@ const txBodies = async (req: Request, res: Response) => {
   const mobilePlatformVersionPrefixes = ["android / ", "ios / ", "- /"];
   const clientVersionHeader = 'yoroi-version';
-  const minMobileVersion = '2.2.2';
+  const minMobileVersion = '2.2.3';
```
@mebassett I think this should be 2.2.4, so 2.2.3 is still shown in maintenance

SebastienGllmt replied:
yeah either that or we can check the sermver check to be <= instead of <

nicarq commented:
@mebassett this will work for mobile. For Desktop are you thinking of doing a new PR?

mebassett commented:
> @mebassett this will work for mobile. For Desktop are you thinking of doing a new PR?

no, this won't work for desktop.  actually that's trickier.  for me, my production chrome extension is sending `yoroi-version: chrome / 0.0.0`.   I'd prefer to resolve that in this PR.  can someone confirm that later versions of yoroi are sending out a non-zero string? 

(it doesn't really matter, because a quick semverCompare would still put it into maintenance mode.  but I also want to be able to come out of maint mode.)

Commit:
add - maintence version checker for desktop.

1. if the client doesn't give us a platform in the version, we assume it is mobile.
2. Locally, my build of yoroi-frontend sends `chrome / 0.0.0`, and in later requests
   sends `chrome / 3.0.0`, so it goes into maint mode when I first load it but then
   comes out of it.  This is probably not desired behavior.
3. 3.0.0 > 1.10.3, the latter being the version num I was asked to implement.

SebastienGllmt commented:
Ah yes, this is because a very old bug in Yoroi where Yoroi instead sends the version you last opened Yoroi with instead of the version you're currently using. That means it sends 0.0.0 the very first time you launch Yoroi.

mebassett commented:
> Ah yes, this is because a very old bug in Yoroi where Yoroi instead sends the version you last opened Yoroi with instead of the version you're currently using. That means it sends 0.0.0 the very first time you launch Yoroi.

what should we do about this?  It does not sound as though its a big deal, as the app will go into maint mode now and come out of it after they upgrade (eventually...)   Should we ignore it for this PR

can you confirm the desired version number for the desktop extensions?  The version I'm seeing in my production extension (after the `0.0.0`) is `1.10.3` but locally with a development build I'm getting `3.0.0`.

SebastienGllmt commented:
The next version we're releasing is 3.0.0
Yes, it doesn't matter if 0.0.0 is set to maintenance since we'll remove the maintenance check after the new version is deployed to all users.

mebassett commented on src/index.ts:
```diff
@@ -240,8 +240,10 @@ const txBodies = async (req: Request, res: Response) => {
   const clientVersionHeader = 'yoroi-version';
-  const minMobileVersion = '2.2.2';
+  const minMobileVersion = '2.2.4';
+  const minDesktopVersion = '1.10.4';
```
should this be `3.0.0` then?

SebastienGllmt replied:
either `1.10.4` or `3.0.0` should work equally fine.

mebassett replied:
okay.  anything else you'd like to see before merging-and-deploying ?
